packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -725,6 +736,56 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     return workInProgress.stateNode;
   }
 
+  function updateTimeoutComponent(
+    current,
+    workInProgress,
+    renderExpirationTime,
+  ) {
+    if (enableSuspense) {
+      const nextProps = workInProgress.pendingProps;
+      const prevProps = workInProgress.memoizedProps;
+
+      const prevDidTimeout = workInProgress.memoizedState;
+
+      // The update queue is only used to store expired promises, and to
+      // schedule a re-render once an expired promise resolves. It does not
+      // determine whether we should show the placeholder state, because we
+      // always attempt to show the placeholder state on every render.
+      const updateQueue = workInProgress.updateQueue;
+      if (updateQueue !== null) {
+        processUpdateQueue(
+          workInProgress,
+          updateQueue,
+          null,
+          null,
+          renderExpirationTime,
+        );
```

Comment:
I feel like the update queue is the wrong data structure here. It's too complex for what it is being used for.

To the point where I don't really know what this is doing or if it is really doing anything.